src/field.jl

```diff
@@ -375,11 +408,12 @@ fromtype(::Type{StrRange}) = StringToken(StrRange)
     unsafe_string(pointer(str, 1 + r.offset), r.length)
 end
 
-@inline function _substring(::Type{StrRange}, str, i, j)
-    StrRange(i - 1, j - i + 1)
+@inline function _substring(::Type{StrRange}, str, i, j, escapecount, escapechar, quotechar, includequotes)
+    StrRange(i - 1, j - i + 1, escapecount)
 end
 
-@inline function _substring(::Type{<:WeakRefString}, str, i, j)
+@inline function _substring(::Type{<:WeakRefString}, str, i, j, escapecount, escapechar, quotechar, includequotes)
+    escapecount > 0 && error("Not yet handled.")
```

Comment:
I think this is mostly called for detecting NA strings? Or somewhere else? I'd be tempted to leave the error message in here until we find a situation where this actually needs to handle escaping.